gossip/service.go

```diff
@@ -166,7 +166,8 @@ func (s *Service) processEvent(realEngine Consensus, e *inter.Event) error {
 		s.occurredTxs.Clear()
 	}
 
-	return nil
```

Comment:
check lines 137-143. 
- Add handling for case of realEngine == nil; otherwise error in line 161 (realEngine.GetEpoch()).
- is it possible to check if err occurs from ProcessEvent first before SetEvent? The SetEvent / DeleteEvent pair seems not ideal.